wpilibj/src/main/java/edu/wpi/first/wpilibj/experimental/command/ParallelCommandGroup.java

```diff
@@ -0,0 +1,92 @@
+package edu.wpi.first.wpilibj.experimental.command;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import edu.wpi.first.wpilibj.command.IllegalUseOfCommandException;
+
+public class ParallelCommandGroup extends CommandGroupBase implements Command {
+
+  //maps commands in this group to whether they are still running
+  private final Map<Command, Boolean> m_commands = new HashMap<>();
+
+  /**
+   * Creates a new ParallelCommandGroup.  The given commands will be executed simultaneously.
+   * The command group will finish when the last command finishes.  If the commandgroup is
+   * interrupted, only the commands that are still running will be interrupted.
+   *
+   * @param commands the commands to include in this group.
+   */
+  public ParallelCommandGroup(Command... commands) {
+    if (!Collections.disjoint(Set.of(commands), getGroupedCommands())) {
+      throw new IllegalUseOfCommandException("Commands cannot be added to multiple CommandGroups");
+    }
+
+    registerGroupedCommands(commands);
+
+    for (Command command : commands) {
+      m_commands.put(command, true);
+    }
+  }
+
+  @Override
+  public void initialize() {
+    for (Command command : m_commands.keySet()) {
```

Comment:
Shouldn't these be using entrySet() to avoid actually needing the hash lookup with the replace? In all instances, would avoid all hash lookups. In addition, maybe this should just be using a list of tuples to avoid needing to hash at all. The only case this would make harder would be isFinished, but I think that would be ok.

Reply:
Made the requested change.  I think it nontrivially harms readability, though.

Reply:
I think we could get some of the readability back by storing the Command in a temporary reference copy, because I agree all the getKey() calls look pretty bad. But a single Command reference is free and would clean up looks a lot.

Reply:
I just tried a different name for the entry, which I think works reasonably well.  Thoughts?